tests/pipeline/test_transform_cis_soc_data.py

```diff
@@ -7,22 +7,45 @@ def test_transform_cis_soc_data(spark_session):
     input_df = spark_session.createDataFrame(
         data=[
             ("JOB TITLE 1", "JOB1", 5),
-            ("JOB TITLE 1", "JOB1", 5),
-            ("JOB TITLE 2", "JOB2", 6),
+            ("JOB TITLE 1", "JOB1", 5),  # Complete duplicates get deduplicated
+            ("JOB TITLE 2", "JOB2", 6),  # Less specific, so this is dropped
             ("JOB TITLE 2", "JOB2", 66),
-            ("JOB TITLE 2", "JOB2", 77),
-            ("JOB TITLE 2", "JOB2", "un"),
+            ("JOB TITLE 2", "JOB2", 77),  # Two codes with equally high specificity, so require manual resolution
+            ("JOB TITLE 2", "JOB2", "un"),  # Duplicates have codes, so this is dropped
             ("JOB TITLE 3", "JOB3", "un"),
             (None, "JOB3", "un"),
+            ("JOB TITLE 4", "JOB4", ""),
+            ("JOB TITLE 5", "JOB5", ""),  # Duplicates have codes, so this is dropped
+            ("JOB TITLE 5", "JOB5", 7),
+            ("JOB TITLE 5", "JOB5", 100),  # One code with higher specificity, so can be automatically resolved
             (None, None, None),
         ],
         schema="work_main_job_title string, work_main_job_role string, standard_occupational_classification_code string",
     )
     expected_df = spark_session.createDataFrame(
-        data=[("JOB TITLE 1", "JOB1", 5), ("JOB TITLE 2", "JOB2", 66), ("JOB TITLE 2", "JOB2", 77)],
-        schema="work_main_job_title string, work_main_job_role string, standard_occupational_classification_code string",
+        data=[
+            ("JOB TITLE 1", "JOB1", 5, False),
+            ("JOB TITLE 3", "JOB3", "uncodeable", True),
+            ("JOB TITLE 4", "JOB4", "uncodeable", True),
+            ("JOB TITLE 5", "JOB5", 100, False),
+        ],
+        schema="work_main_job_title string, work_main_job_role string, standard_occupational_classification_code string, soc_code_edited_to_uncodeable boolean",
+    )
+    expected_conflicts_df = spark_session.createDataFrame(
+        data=[
+            ("JOB TITLE 2", "JOB2", 66, "AMBIGUOUS AFTER DEDUPLICATION", False),
+            ("JOB TITLE 2", "JOB2", 77, "AMBIGUOUS AFTER DEDUPLICATION", False),
+            ("JOB TITLE 2", "JOB2", 6, "NOT MOST SPECIFIC", False),
```

Comment:
"JOB2" records 6 and "un" aren't conflicts because we've resolved them automatically. The user only needs to see the 77 and 66 records to manually resolve this.
For "JOB5", we've automatically resolved the conflict to get the 100 records, so there are no remaining conflicts to report in this table